hoomd/pytest/test_device.py

```diff
@@ -108,3 +108,33 @@ def test_cpu_build_specifics():
         pytest.skip("Don't run CPU-build specific tests when GPU is available")
     assert not hoomd.device.GPU.is_available()
     assert type(hoomd.device.auto_select()) == hoomd.device.CPU
+
+
+def test_device_notice(device, tmp_path):
+    # Message file declared. Should output in specified file.
+    device.notice_level = 4
+    device.msg_file = str(tmp_path / "str_message")
+    msg = "This message should output."
+    device.notice(msg)
+
+    if device.communicator.rank == 0:
+        with open(device.msg_file) as fh:
+            assert fh.read() == msg + "\n"
```

Comment:
My only concern is different end of file conventions on different platforms. Would that sometimes cause the test to fail if it doesn't end with a newline?

Reply:
This is not a problem. See https://docs.python.org/3/library/functions.html#open
> When reading input from the stream, if newline is None, universal newlines mode is enabled. Lines in the input can end in '\n', '\r', or '\r\n', and these are translated into '\n' before being returned to the caller. If it is '', universal newlines mode is enabled, but line endings are returned to the caller untranslated. If it has any of the other legal values, input lines are only terminated by the given string, and the line ending is returned to the caller untranslated.